File: dotnet/DesignPatterns/BehavioralDesignPatterns/StateDesignPattern/StateDesignPattern.Example1/CancelledState.cs
namespace StateDesignPattern.Example1;

/// <summary>
/// Concrete State - represents a cancelled order.
/// </summary>
public class CancelledState : IOrderState
{
    public void ProcessOrder(Order order)
    {
        Console.WriteLine("Cannot process. Order has been cancelled.");
    }

    public void ShipOrder(Order order)
    {
        Console.WriteLine("Cannot ship. Order has been cancelled.");
    }

    public void DeliverOrder(Order order)
    {
        Console.WriteLine("Cannot deliver. Order has been cancelled.");
    }

    public void CancelOrder(Order order)
    {
        Console.WriteLine("Order has already been cancelled.");
    }

    public string GetStateName() => "Cancelled";
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/StateDesignPattern/StateDesignPattern.Example1/src/CancelledState.ts
import { IOrderState } from './IOrderState';
import { Order } from './Order';

export class CancelledState implements IOrderState {
  processOrder(order: Order): void {
    console.log('Cannot process. Order has been cancelled.');
  }

  shipOrder(order: Order): void {
    console.log('Cannot ship. Order has been cancelled.');
  }

  deliverOrder(order: Order): void {
    console.log('Cannot deliver. Order has been cancelled.');
  }

  cancelOrder(order: Order): void {
    console.log('Order has already been cancelled.');
  }

  getStateName(): string {
    return 'Cancelled';
  }
}
